service/worker/sysworkflow/constants.go

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package sysworkflow
+
+// RequestType is the type for signals that can be sent to system workflows
+type RequestType int
+
+const (
+	// Domain is the cadence system workflows domain
+	Domain = "cadence-system"
+
+	// DecisionTaskList is the task list that all system workflows share
+	DecisionTaskList = "cadsys-decision-tl"
+
+	// SignalName is the name of the cadence signal that system tasks are sent on
+	SignalName = "cadsys-signal-sig"
+
+	// WorkflowIDPrefix is the prefix of all system workflow ids
+	WorkflowIDPrefix = "cadsys-wf"
+
+	// SignalsUntilContinueAsNew is the number of signals system workflow must receive before continuing as new
+	SignalsUntilContinueAsNew = 1000
```

Comment:
This feels somewhat arbitrary. I think we need to do some load testing to figure out a reasonable value here.